### ISSUE TITLE
Label dataset

### ISSUE THREAD

**Issue by marco-c** [marco-c/autowebcompat#2]

**Comment by sagarvijaygupta** [comment]
@marco-c Could you please elaborate the issue?

**Comment by amit2rockon** [comment]
labels.json file to be loaded with labeled data information of the data.

**Comment by amit2rockon** [comment]
Some of the data are too explicit.  @marco-c  What to label the data which just differ in single character  'y','d'?

**Comment by sagarvijaygupta** [comment]
@amit2rockon Thanks for the clarification. @marco-c  Should we not let more than one person do the labeling? We will get better voting in that case. Should I right code for that? That might even address the conflicting issue @amit2rockon has mentioned.

**Comment by marco-c** [comment]
I've updated the first comment with a more thorough explanation :)

**Comment by marco-c** [comment]
I've opened #20 for the first point and #21 for the second point. This issue will be about the labeling itself.

**Comment by Zzoey** [comment]
@marco-c A CNN learns more about the patterns in the image (Edges, Corners and their correlations) from example 2 it is evident that it will be difficult for a NN to learn the adversary and classify that both are compatible. 

To detect differences, Y+D and N in a better way or even Y and D+N, I think we can focus more on, Finding ROIs (Attention based) and feed those patches to the NN. This can be our next go-to-go (alternative) if nothing works very well after training part which you suggested.

**Comment by nok** [comment]
At the beginning I would start with screenshots based on equal page sources (same content), so only Y. Furthermore I would try to normalise the device settings to bring the rendered Firefox version closer to the rendered Chrome version. And maybe we could remove the system look and feel elements by injecting a small script before the screenshot will be taken.

**Comment by marco-c** [comment]
> At the beginning I would start with screenshots based on equal page sources (same content), so only Y. 

Do you mean Y vs D+N?
N.B.: The screenshots are always based on equal page sources.

> Furthermore I would try to normalise the device settings to bring the rendered Firefox version closer to the rendered Chrome version. And maybe we could remove the system look and feel elements by injecting a small script before the screenshot will be taken.

This is #13.

**Comment by nok** [comment]
Yes, I meant Y vs D+N.

Thanks, today I don't have an overview over the issues and tasks.

**Comment by marco-c** [comment]
> Thanks, today I don't have an overview over the issues and tasks.

No worries! And keep the suggestions coming ;)
It's better if you file a new issue for every idea you have. In the worst case we close it as a duplicate.

**Comment by Shashi456** [comment]
@marco-c i'd like to label parts of our dataset, how do you suggest i go about doing that ? because as far as i've seen there is no script which merges labels from the label_persons directory into the actual labels directory .

**Comment by sagarvijaygupta** [comment]
@Shashi456 I think you are talking about `generate_labels.py`.

**Comment by Shashi456** [comment]
@sagarvijaygupta oh , i thought it wasn't updated for the new files :P

**Comment by marco-c** [comment]
> @marco-c i'd like to label parts of our dataset, how do you suggest i go about doing that ? because as far as i've seen there is no script which merges labels from the label_persons directory into the actual labels directory .

The script hasn't been updated yet to deal with bounding boxes, but you can already start labeling and pushing your labels file to the repo. Then, once we have the script done, we will actually combine the labeling done by you and the labeling done by other persons.

**Comment by sdv4** [comment]
I am running label.py on my mac, and I am finding that it is slow or unresponsive on non-y images. For instance, it takes a long from when I try to drop a boundary box to when it shows up and for the 'T', resizing arrow, and movement arrow show up. Clicking on any causes everything to disappear until I release my mouse + a couple of seconds. 

Is this a problem that anyone else has come up against?

**Comment by marco-c** [comment]
It could be a Mac issue, I think nobody has tested it on a Mac yet. Could you try in a Linux VM?

**Comment by sdv4** [comment]
@marco-c I am not having that problem on the Linux VM, so I can label a lot faster now. A couple of questions:

- Applying labels: suppose two images seem to only be different in terms of the position on the page that has been scrolled to (ex. Image 1 looks like image 2, except that image 2 has been scrolled down and thus exposes more of the page content). Would these be considered compatible, not compatible, or compatible but different.

- Getting my labels into the main repo: Should I open a PR for a new branch off of my forked master that is the same as the upstream master, except that it includes my new labels?

**Comment by sdv4** [comment]
Also, how would you label a pair of images when they show the same page except that one is in English and the other in Italian?

**Comment by sagarvijaygupta** [comment]
@sdv4 you can take help from the #220 till it is merged. Those screenshots are marked by @marco. 
For the last one you should mark them incompatible while drawing bounding box on Italian side.

**Comment by marco-c** [comment]
> Getting my labels into the main repo: Should I open a PR for a new branch off of my forked master that is the same as the upstream master, except that it includes my new labels?

Yes! You can open a PR that says "Add some labels from Shane Sims".

**Comment by marco-c** [comment]
Are the other two questions answered by #220?

**Comment by sagarvijaygupta** [comment]
@marco-c For the scroll one we have marked them as incompatible in screenshots, and for italian one we mark bounding boxes in italian side with incompatibility.

**Comment by marco-c** [comment]
> For the scroll one we have marked them as incompatible in screenshots

IIRC I've marked them as compatible, didn't I?

**Comment by marco-c** [comment]
No maybe not, they should be incompatible (e.g. if clicking on a button causes a scroll in one browser, it should cause a scroll in the other browser too).

**Comment by sagarvijaygupta** [comment]
https://github.com/marco-c/autowebcompat/blob/b18eae0999b6389b1cc84153f43b077eddccd9d8/collect.py#L162

And if this script works differently on two browsers then also it should be an incompatibility?

**Comment by marco-c** [comment]
> And if this script works differently on two browsers then also it should be an incompatibility?

It shouldn't, but it's hard to tell whether it was this script that failed or something else.
Maybe we should just assume this always works.

**Comment by sagarvijaygupta** [comment]
Okay!

**Comment by Shashi456** [comment]
@marco-c @sagarvijaygupta  so while i was labeling the dataset one of the major themes that popped up was how chrome had a scrollbar. Almost all images which have a scrollbar are very similar but the scrollbars adds a shift which makes the overlay look incompatible . 

Should we update the crawler options for chorme to remove the scroll bar or suggest the user something accordingly in the labeling guide?

**Comment by sagarvijaygupta** [comment]
@Shashi456 it is already removed from the crawler.
